app/models/manageiq/providers/nuage/network_manager/vsd_client.rb

```diff
@@ -78,8 +78,7 @@ def get_domains_for_enterprise(enterprise_id)
     end
 
     def get_zones
-      @rest_call.append_headers("X-Nuage-FilterType", "predicate")
-      @rest_call.append_headers("X-Nuage-Filter", "name ISNOT 'BackHaulZone'")
+      exclude_name('BackHaulZone')
```

Comment:
seems like the whole function can be extracted to something like

get_list('zones')
get_list('subnets') 
etc.

since the collection name seems to be the only difference

Reply:
Although I agree we can have a `get_list` that will return the collection, I would nevertheless prefer that we keep `get_zones` and `get_subnets` that in turn call `get_list`. Otherwise, people need to know exactly what to pass as arguments.

@Ladas would this be ok for you?

Similarly we could update `get_zone` and `get_subnet` that call just `get_object(type, id)`.

Reply:
yes, it's just about abstraction the similar code to one place, so 

```
def get_zones
   get_list('zones')
end
```

Reply:
@miha-plesko Since you started this, can you please cleanup the code as @Ladas suggested? It should be a fairly trivial change and will also make codeclimate happy.